### ISSUE TITLE
`missing_errors_doc` dont work when attribute `#[profiling::function]` is set

### ISSUE THREAD

**Issue by gwen-lg** [rust-lang/rust-clippy#13484]
## Summary
When attribute `#[profiling::function]` is set on a function, the lint `missing_errors_doc` trigger any warning even if the function can return an error and possible returned `Errors` is not documented.
While just remove the attribute allow to have report of the missing `Error` doc again.

## Lint Name

missing_errors_doc

### trigger the lint as expected
```rust
/// Divide 100 by the value indicated
/// => Reference which trigger the lint.
pub fn div_hundred_by(value: u32) -> Result<u32, Error> {
    if value == 0 {
        return Err(Error::DivByZero);
    }
    Ok(100 / value)
}
```
### don't trigger the lint
```rust
/// Divide 1000 by the value indicated
/// => variation with marker `#[profiling::function]` added.
#[profiling::function]
pub fn div_thousand_by(value: u32) -> Result<u32, Error> {
    if value == 0 {
        return Err(Error::DivByZero);
    }
    Ok(1000 / value)
}

```
A demo project is available here : https://github.com/gwen-lg/clippy-lints-doc-tests
